python/lsst/afw/cameraGeom/basicCameraFactory.py

```diff
@@ -0,0 +1,225 @@
+    Det01 0.0 9000.0 2.0 400 400 science 10.0 0.0 0.0
+    Det02 -9000.0 0.0 2.0 400 400 wave 30.0 0.0 0.0
+    """
+
+    def __init__(self, detectorLayoutFile,
+                 detectorIdFromAbbrevName=None,
+                 detTypeMap=None,
+                 radialTransform=[0.0, 1.0, 0.0, 0.0],
+                 cameraName='LSST'
+                 ):
+
+        """
+        @param [in] detectorLayoutFile is the absolute path to the file
+        listing the layout of all of the chips in the camera's focal plane
+        (see this class' docstring for detailed information on the contents
+        of that file).
+
+        @param [in] detectorIdFromAbbrevName is a (required) method mapping the
+        names of detectors as stored in the detectorLayoutFile to unique integers
+        identifying each detector.
+
+        @param[in] detTypeMap is a dict mapping the group name of the sensors in
+        detectorLayoutFile to the integers:
+
+        0 = SCIENCE
+        1 = FOCUS
+        2 = GUIDER
+        3 = WAVEFRONT
+
+        if left as None, this will just try to cast the contents of detectorLayoutFile
+        as an int
+
+        @param [in] radialTransform is a list of coefficents that transform positions
+        from pupil coordinates (x, y, radians) to focal plane coordinates (x, y, mm)
+        according to the convention
+
+        (x, y)_focal = (x, y)_pupil * sum_i radialTransform[i] * r^i/r
+
+        where r is the magnitude of the point in pupil coordinates.  Note that the [1]
+        element of this list should be 1/plateScale where plateScale is in radians per mm.
+
+        @param [in] cameraName is a string referring to the name of the camera
+        (default 'LSST')
+        """
+
+        self._detectorLayoutFile = detectorLayoutFile
+        self._shortNameFromLongName = {}
+
+        self._cameraName = cameraName
+        self._radialTransform = radialTransform #[1] should be 1/rad per mm
+
+        if detectorIdFromAbbrevName is None:
+            raise RuntimeError("You must specify detectorIdFromAbbrevName when "
+                               "instantiating BasicCameraFactory")
+        else:
+            self._detectorIdFromAbbrevName = detectorIdFromAbbrevName
+
+        self._detTypeMap = detTypeMap
+
+        self._camConfig = None
+        self._ampTableDict = None
+
+
+    def detTypeMap(self, typeName):
+        """
+        Map the strings indicating detector type in the focal plane layout file
+        to the integers used by afw.cameraGeom to keep track of detector types.
+        """
+        if self._detTypeMap is None:
+            return int(typeName)
+        else:
+            return self._detTypeMap[typeName]
+
+    def makeDetectorConfigs(self):
+        """
+        Create the detector configs to use in building the Camera
+        """
+        detectorConfigs = []
+        #We know we need to rotate 3 times and also apply the yaw perturbation
+        nQuarter = 1
```

Comment:
Is it that the LSST configuration assumes a baseline 90 degree rotation in addition to the yaw specified in the configuration file?

Should we just get rid of nQuarter altogether and have the full yaw rotation be specified in the configuration file?